kconfig: BT_SEND_ECC_EMULATION does not select BT_LONG_WQ

**Describe the bug**

Commit https://github.com/zephyrproject-rtos/zephyr/commit/38e799fe2dd6a34009f0779b21f0d848d8823bfd enabled `BT_SEND_ECC_EMULATION` for all silabs boards by default, but `BT_LONG_WQ` is not being selected, even though it should be selected according to `subsys/bluetooth/host/Kconfig`: https://github.com/zephyrproject-rtos/zephyr/blob/595d81a941c5bdcb67d7d33c9125d0aba2194a6a/subsys/bluetooth/host/Kconfig#L1010

As a result Bluetooth-enabled applications will no longer compile for all boards under `boards/silabs`. 

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->
1. Checkout https://github.com/zephyrproject-rtos/zephyr/commit/38e799fe2dd6a34009f0779b21f0d848d8823bfd
2. `west build -b xg24_dk2601b --pristine always samples/bluetooth/observer`
3. Observe the error below
```
/opt/toolchain/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(hci_ecc.c.obj): in function `le_gen_dhkey':
/home/user/src/zephyr/zephyr/subsys/bluetooth/host/hci_ecc.c:301: undefined reference to `bt_long_wq_submit'
/opt/toolchain/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(hci_ecc.c.obj): in function `le_p256_pub_key':
/home/user/src/zephyr/zephyr/subsys/bluetooth/host/hci_ecc.c:341: undefined reference to `bt_long_wq_submit'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
4. Observe `# CONFIG_BT_LONG_WQ is not set` in `.config`
5. Manually set `CONFIG_BT_LONG_WQ=y` in `.config`, run `west build -b xg24_dk2601b samples/bluetooth/observer`
6. Build is now successful

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->
`CONFIG_BT_LONG_WQ` should be selected automatically.

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->
Bluetooth-enabled applications will no longer compile for silabs boards by default.

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA or Version used: 38e799fe2dd6a34009f0779b21f0d848d8823bfd

--- a/boards/silabs/dev_kits/sltb010a/Kconfig.defconfig
+++ b/boards/silabs/dev_kits/sltb010a/Kconfig.defconfig
@@ -26,9 +26,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 config REGULATOR

--- a/boards/silabs/dev_kits/xg24_dk2601b/Kconfig.defconfig
+++ b/boards/silabs/dev_kits/xg24_dk2601b/Kconfig.defconfig
@@ -31,9 +31,6 @@ config COMMON_LIBC_MALLOC_ARENA_SIZE
 config MAIN_STACK_SIZE
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 if SHELL
 
 config SHELL_STACK_SIZE

--- a/boards/silabs/dev_kits/xg27_dk2602a/Kconfig.defconfig
+++ b/boards/silabs/dev_kits/xg27_dk2602a/Kconfig.defconfig
@@ -26,9 +26,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 config REGULATOR

--- a/boards/silabs/radio_boards/slwrb4104a/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4104a/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4104A

--- a/boards/silabs/radio_boards/slwrb4161a/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4161a/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4161A

--- a/boards/silabs/radio_boards/slwrb4170a/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4170a/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4170A

--- a/boards/silabs/radio_boards/slwrb4180a/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4180a/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4180A

--- a/boards/silabs/radio_boards/slwrb4250b/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4250b/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4250B

--- a/boards/silabs/radio_boards/slwrb4255a/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/slwrb4255a/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_SLWRB4255A

--- a/boards/silabs/radio_boards/xg24_rb4187c/Kconfig.defconfig
+++ b/boards/silabs/radio_boards/xg24_rb4187c/Kconfig.defconfig
@@ -37,9 +37,6 @@ config MAIN_STACK_SIZE
 	default 3072 if PM
 	default 2304
 
-config BT_SEND_ECC_EMULATION
-	default y
-
 endif # BT
 
 endif # BOARD_XG24_RB4187C

--- a/drivers/bluetooth/hci/Kconfig
+++ b/drivers/bluetooth/hci/Kconfig
@@ -126,6 +126,7 @@ config BT_SILABS_EFR32
 	select MBEDTLS
 	select MBEDTLS_PSA_CRYPTO_C
 	select MBEDTLS_ENTROPY_C
+	select HAS_BT_CTLR
 	help
 	  Use Silicon Labs binary Bluetooth library to connect to the
 	  controller.

